src/app/composer/qgscomposermanager.cpp

```diff
@@ -110,14 +115,61 @@ void QgsComposerManager::refreshComposers()
   mComposerListWidget->sortItems();
 
   // Restore selection
-  if ( !selName.isEmpty() )
+  if ( !selectedComposers.isEmpty() )
   {
-    QList<QListWidgetItem*> items = mComposerListWidget->findItems( selName, Qt::MatchExactly );
-    if ( !items.isEmpty() )
+    Q_FOREACH ( QgsComposer* c, selectedComposers )
```

Comment:
This can be simplified a bit - I'd suggest making `selectedComposers` a QSet, and then doing something like:

    while ( i != mItemComposerMap.constEnd() )
     {
	// This composer was selected: reselect it !
	if ( selectedComposers.contains( i.value() ) )
	{
	 ...
	}
	++i;
      }

that would avoid the need for the outer `Q_FOREACH ( QgsComposer* c, selectedComposers )` loop